### PR TITLE
fulltext search on work landing page

### PR DESCRIPTION
**Pull request by davidgrove73**

### CODE REVIEW

File: app/assets/javascripts/catalog.js [hunk 1]
```diff
@@ -22,10 +22,41 @@ $(document).ready(function(){
...
+        target_selector = $(this).data('target');
+        q = encodeURIComponent($(qselector).val());
+        $.ajax({
+            type: 'GET',
+            url: '/catalog.json?search_field=leaf&rows=200&q='+q+'&workid='+workid,
+            datatype: 'json',
+            success: function(data) {
+                $(target_selector).empty();
+                docs = data.response.docs
+                $(target_selector).append('<div id="results-header"><p>'+data.response.pages.total_count+' Matches</p></div>');
```
Comment (ronan-mch) [inline app/assets/javascripts/catalog.js:42]: Matches -> Fund?

File: app/models/search_builder.rb [hunk 2]
```diff
@@ -1,3 +1,12 @@
...
+
+  def add_work_id solr_params
+    if blacklight_params[:search_field] == 'leaf' && blacklight_params[:workid].present?
+      solr_params[:fq] ||= []
+      workid = blacklight_params[:workid]
+      workid = "#{workid}*" unless workid.include? '*'
+      solr_params[:fq] << "part_of_ssi:#{workid}"
+    end
+  end
 end
```
Comment (ronan-mch) [inline app/models/search_builder.rb:12]: I think you could put this method in the CatalogController as well, but perhaps it's better practice to put it here? In that case @xenma's  branch should be refactored a bit after a merge to put the search_params logic in the same place...

**Comment by ronan-mch** [comment]
:+1:  :+1:  :+1: Excellent!
Shouldn't we be able to link from a catalog search results page into a search on the work landing page? E.g., if I search for Nat and click on Natten by Aarestrup, the internal search on the landing page should already be populated with results? This might mean passing the query term as a url param to the work landing page; the javascript could then check for such a parameter on page load and initiate a search if it's present.

**Comment by davidgrove73** [comment]
added js-code to automatically trigger a search on the work landing page

**Pull request merged**